File: src/main/java/axirassa/overlord/MonitorPeriod.java

package axirassa.overlord;

import lombok.Getter;

/**
 * Represents a period of time over which the number of restarts of a process
 * may be watched.
 * 
 * @author wiktor
 * 
 */
public enum MonitorPeriod {
	MINUTE(60 * 1000),
	HOUR(60 * 60 * 1000),
	DAY(24 * 60 * 60 * 1000);

	@Getter
	private final long millis;


	MonitorPeriod (long millis) {
		this.millis = millis;
	}
}

File: src/main/java/axirassa/overlord/MonitorPeriod.scala

package axirassa.overlord

/**
 * Represents a period of time over which the number of restarts of a process
 * may be watched.
 *
 * @author wiktor
 *
 */
object MonitorPeriod {
  val MINUTE = new MonitorPeriod(60 * 1000)
  val HOUR = new MonitorPeriod(60 * 60 * 1000)
  val DAY = new MonitorPeriod(24 * 60 * 60 * 1000)
}

class MonitorPeriod(var millis : Long)